apps/molgenis-components/src/client/client.ts

```diff
@@ -232,21 +248,43 @@ const deleteAllTableData = (tableId: string, schemaId?: string) => {
 
 const fetchSchemaMetaData = async (
   schemaId?: string
-): Promise<ISchemaMetaData> => {
-  const currentschemaId = schemaId ? schemaId : "CACHE_OF_CURRENT_SCHEMA";
-  if (schemaCache.has(currentschemaId)) {
-    return schemaCache.get(currentschemaId) as ISchemaMetaData;
+): Promise<Record<string, ISchemaMetaData>> => {
+  const currentSchemaId = schemaId ? schemaId : "CACHE_OF_CURRENT_SCHEMA";
+  if (schemaCache.has(currentSchemaId)) {
+    return Object.fromEntries(schemaCache);
   }
   return await axios
     .post(graphqlURL(schemaId), { query: metaDataQuery })
-    .then((result: AxiosResponse<{ data: { _schema: ISchemaMetaData } }>) => {
-      const schema = result.data.data._schema;
-      if (schemaId == null) {
-        schemaCache.set(currentschemaId, schema);
+    .then(
+      async (result: AxiosResponse<{ data: { _schema: ISchemaMetaData } }>) => {
+        const schema = result.data.data._schema;
+        const cacheKey = schemaId ? schemaId : currentSchemaId;
+        schemaCache.set(cacheKey, schema);
+
+        // create if list of linked schemas
+        const linkedSchemasIds = schema.tables.reduce(
+          (accum: string[], table: ITableMetaData) => {
+            table.columns.forEach((column: IColumn) => {
+              if (column.refSchemaId && !accum.includes(column.refSchemaId)) {
+                accum.push(column.refSchemaId);
+              }
+            });
+            return accum;
+          },
+          []
+        );
+
+        // fetch metaData for all linked schemas
+        await Promise.all(
```

Comment:
ermm, if we do this here is it then not better to reinstate the server side code? What do you think?

More importantly, are you doing this recursively? Because linked schema can have linked schema also?

Reply:
both server and client have pros and cons ( imo most of the dynamic stuff would be better suited to a rest like interface ( this is building the rest struct client side) 

 its recursive, due to  fetchSchemaMetaData on line 280 ( need to maybe add a test for this )